plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/utils/ProjectImportEnablingScope.java

```diff
@@ -87,8 +88,10 @@
 	 *            {@link IEObjectDescriptionWithError} will be returned instead of <code>null</code> in case of
 	 *            unresolvable references).
 	 */
-	public static IScope create(IN4JSCore n4jsCore, Resource resource, Optional<ImportDeclaration> importDecl,
+	public static IScope create(IN4JSCore n4jsCore, N4JSModel n4jsModel, Resource resource,
+			Optional<ImportDeclaration> importDecl,
```

Comment:
I like the idea of a clear structure. However, currently the model is injected at 8 places in our codebase. Maybe we should perform a refactoring so that the model is encapsulated by the core? In this case, the model could even be of package local accessibility and/or not use injection.